internal/indexcoord/index_coord.go

```diff
@@ -596,38 +543,70 @@ func (i *IndexCoord) watchMetaLoop() {
 	}
 }
 
-func (i *IndexCoord) assignTasksServerStart() error {
-	sessions, _, err := i.session.GetSessions(typeutil.IndexNodeRole)
-	if err != nil {
-		return err
-	}
-	for _, session := range sessions {
-		addrs := strings.Split(session.Address, ":")
-		ip := addrs[0]
-		port, err := strconv.ParseInt(addrs[1], 10, 64)
-		if err != nil {
-			return err
-		}
+func (i *IndexCoord) assignTaskLoop() {
+	ctx, cancel := context.WithCancel(i.loopCtx)
 
-		req := &indexpb.RegisterNodeRequest{
-			Address: &commonpb.Address{
-				Ip:   ip,
-				Port: port,
-			},
-			NodeID: session.ServerID,
-		}
-		if err = i.addNode(session.ServerID, req); err != nil {
-			log.Debug("IndexCoord", zap.Any("IndexCoord start find node fatal, err = ", err))
+	defer cancel()
+	defer i.loopWg.Done()
+
+	timeTicker := time.NewTicker(durationInterval)
```

Comment:
10 seconds is too long.
please change to 3 seconds.
In the future, a calculation formula can be designed to dynamically modify the number of tasks assigned each time and the timer interval according to the total task load, the rate at which nodes complete tasks, the number of nodes, etc.

Reply:
In the future, it is also necessary to add information about the opening time of the task to the meta. You can make scheduling based on this information to minimize the average waiting time of the task.

Reply:
Good suggestion.